### PR TITLE
feat: Add GuildMapScreen

### PR DESCRIPTION
**Pull request by kristofbolyai**

### CODE REVIEW

**Comment by kristofbolyai** [comment]
image
image
image

File: common/src/main/java/com/wynntils/features/user/map/GuildMapFeature.java [hunk 1]
```diff
@@ -4,15 +4,38 @@
...
 import com.wynntils.wynn.model.territory.GuildTerritoryModel;
 import java.util.List;
+import org.lwjgl.glfw.GLFW;
 
 @FeatureInfo(category = FeatureCategory.MAP)
 public class GuildMapFeature extends UserFeature {
+    public static GuildMapFeature INSTANCE;
+
+    @Config
+    public boolean renderUsingLinear = true;
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/features/user/map/GuildMapFeature.java:25]: This is more of a general, across all map feature config

File: common/src/main/java/com/wynntils/gui/screens/maps/GuildMapScreen.java [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+            for (String tradingRoute : territoryPoi.getTerritoryInfo().getTradingRoutes()) {
+                TerritoryPoi routePoi =
+                        GuildTerritoryModel.getGuildTerritoryMap().get(tradingRoute);
+                if (routePoi != null) {
+                    float x = MapRenderer.getRenderX(routePoi, mapCenterX, centerX, currentZoom);
+                    float z = MapRenderer.getRenderZ(routePoi, mapCenterZ, centerZ, currentZoom);
+
+                    RenderUtils.drawLine(poseStack, CommonColors.DARK_GRAY, poiRenderX, poiRenderZ, x, z, 0, 1);
+                }
+            }
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/gui/screens/maps/GuildMapScreen.java:95]: Does this double render?

File: common/src/main/java/com/wynntils/gui/screens/maps/GuildMapScreen.java [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+        }
+
+        // Reverse and Render
+        for (int i = filteredPois.size() - 1; i >= 0; i--) {
+            Poi poi = filteredPois.get(i);
+
+            float poiRenderX = MapRenderer.getRenderX(poi, mapCenterX, centerX, currentZoom);
+            float poiRenderZ = MapRenderer.getRenderZ(poi, mapCenterZ, centerZ, currentZoom);
+
+            poi.renderAt(poseStack, poiRenderX, poiRenderZ, hovered == poi, poiScale, currentZoom);
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/gui/screens/maps/GuildMapScreen.java:105]: hovered isnt set

Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/gui/screens/maps/GuildMapScreen.java:105]: Nevermind

File: common/src/main/java/com/wynntils/gui/screens/maps/GuildMapScreen.java [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+
+        if (territoryInfo.isHeadquarters()) {
+            renderYOffset += 20;
+            FontRenderer.getInstance()
+                    .renderText(
+                            poseStack,
+                            ChatFormatting.RED + "Guild Headquarters",
+                            10,
+                            10 + renderYOffset,
+                            CommonColors.WHITE,
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/gui/screens/maps/GuildMapScreen.java]: Change color here instead?

File: common/src/main/java/com/wynntils/features/user/map/GuildMapFeature.java (hunk 1 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/map/GuildMapFeature.java:25]: I didnt wanna use the other feature's config?

**Comment by magicus** [comment]
I briefly looked through this and found no oddities, but I have not checked it in detail.

File: common/src/main/java/com/wynntils/gui/screens/maps/GuildMapScreen.java (hunk 2 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/gui/screens/maps/GuildMapScreen.java:95]: Yeah but it is faster to double render than to do any calculations optimisations.

Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/gui/screens/maps/GuildMapScreen.java:95]: note this explicitly

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/gui/screens/maps/GuildMapScreen.java:95]: Done

**Review by Incompleteusern — APPROVED** [review]

**Pull request merged**